### PR TITLE
Add httpx support

### LINKED ISSUE #47: Provide `httpx` client
I am not actually using this library, but while I was reading the readme example - I wondered why it is not supported.

`httpx` has both modes: sync and async, which really implies that there should be two integrations / classes for it.

### PR DESCRIPTION
**Pull request by anmayithap**
### Description

This PR implements support for `httpx` as a backend.

### Changes

- **Synchronous Client:** Added support for the standard synchronous `httpx.Client`.
- **Asynchronous Client:** Added support for `httpx.AsyncClient`.
- **Base URL Handling:** Implemented logic to temporarily override `base_url` at the request level (essential for `httpx` compatibility).
- **Tests:** Added integration tests covering basic request scenarios, mirroring the existing coverage for `requests` and `aiohttp`.

### Questions & Suggestions

1. **Type Hints in Tests:**
   I wasn't sure if strict type annotations are required for test files. I noticed that `mypy` raises significant errors when using them with `@pytest_asyncio.fixture` (see the screenshot below). I can certainly add them if needed, please let me know.
   
Снимок экрана 2025-11-24 в 23 38 32

2. **Streaming Support:**
   I considered adding support for `httpx` streaming, but decided to keep this PR focused to avoid scope creep. I am open to implementing it in this PR or a follow-up if you think it's necessary right now.

3. **Documentation:**
   What is the current status/workflow for Sphinx autodoc? Should I include docstrings in the code or add entries to the `.rst` documentation files?

4. **Test Coverage:**
   I have implemented the basic test suite. However, we could extend coverage further, specifically to test edge cases regarding `base_url` overriding logic.

### Linked Issues

Resolves #47

### CODE REVIEW

File: src/descanso/http/httpx.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+        self._base_url = base_url
+        self._session: _HttpxClientT = session
+
+    @contextmanager
+    def _override_base_url(self: Self) -> Iterator[None]:
+        old: URL = self._session.base_url
+
+        if self._base_url is not None:
+            self._session.base_url = self._base_url
```
Comment (Tishka17) [inline src/descanso/http/httpx.py]: it is not thread safe, you can just prepare URL for single request using `url=urllib.parse.urljoin(self._base_url, request.url)` as it is done in other clients

File: src/descanso/http/httpx.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+        yield
+
+        if old != self._session.base_url:
+            self._session.base_url = old
+
+    def _to_httpx_query_params(
```
Comment (Tishka17) [inline src/descanso/http/httpx.py]: I do not see any profit of inheritance here. Usage of generics look like over-complication. Let's make these 2 methods simple functions and get rid of base class

File: src/descanso/http/httpx.py (hunk 1 above)
Comment (anmayithap) [reply src/descanso/http/httpx.py]: Agreed, modifying the session is not thread-safe. I'll revert that.
My concern with urllib.parse.urljoin was that it swallows the last path segment of base_url if a trailing slash is missing (e.g., .../api/v1 + ping becomes .../api/ping).
I will switch to urljoin but I'll add a check to ensure base_url ends with / to prevent this behavior.

File: src/descanso/http/httpx.py (hunk 2 above)
Comment (anmayithap) [reply src/descanso/http/httpx.py]: Agreed. I overcomplicated the design at the last minute — it seemed logical at the time, but I see now it's unnecessary.
